src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -355,51 +354,57 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema)(im
     }
   }
 
-  def importPFB(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]],
-                workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest): Future[PerRequestMessage] = {
+  def importPFB(workspaceNamespace: String, workspaceName: String, pfbRequest: PfbImportRequest): Future[PerRequestMessage] = {
 
-    def callArrow = {
-      val gzipPipeline = addHeader (`Accept-Encoding`(gzip)) ~> pipeline ~> decode(Gzip)
-      gzipPipeline { Post(avroToRawlsURL, pfbRequest) }
-    }
+    // initial validation of the presigned url provided as an argument
+    def validatePfbUrl(pfbRequest: PfbImportRequest)(op: URL => Future[PerRequestMessage]) = {
+      val pfbUrl = try {
+        new URL(pfbRequest.url)
+      } catch {
+        case e: Exception => throw new FireCloudExceptionWithErrorReport(ErrorReport(BadRequest, s"Invalid URL: ${pfbRequest.url}", e))
+      }
+      val acceptableProtocols = Seq("https") //for when we inevitably change our mind and need to support others
 
-    def callRawls(entity: HttpEntity) = {
-      val authPipeline = authHeaders(requestContext) ~> pipeline
-      authPipeline {
-        Post(FireCloudDirectiveUtils.encodeUri(Rawls.entityPathFromWorkspace(workspaceNamespace, workspaceName) + "/batchUpsert"), entity)
+      if (!acceptableProtocols.contains(pfbUrl.getProtocol)) {
+        throw new FireCloudExceptionWithErrorReport(ErrorReport(BadRequest, "Invalid URL protocol: must be https only"))
       }
+
+      // TODO: add real (semantic) validation for origins, to ensure caller isn't supplying malice
+      op(pfbUrl)
```

Comment:
this `TODO` is a stub to reflect Doug's comments in the design doc that we should check hostname or other creds on the signed URL to make sure it looks like what we expect from Gen3.